sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/PodSpecsCannotChangeNetworkRegime.java

```diff
@@ -0,0 +1,88 @@
+        Map<String, PodSpec> newPods = pair.getValue();
+        if (newPods.isEmpty()) {
+            return errors;
+        }
+
+        // check the PodSpecs to make sure none of them make a transition form a state where they use host ports
+        // to one where they don't (or vice versa).
+
+        Map<String, PodSpec> oldPods = nullableOldConfig.getPods().stream()
+                .collect(Collectors.toMap(PodSpec::getType, podSpec -> podSpec));
+        for (Map.Entry<String, PodSpec> kv: newPods.entrySet()) {
+            PodSpec newPod = kv.getValue();
+            // first we check that the new pod's networks (if present) do not erroneously use port mapping
+            String offendingNetworkNames = podSpecOverlayNetworkIsCorrect(newPod);
+            if (!Strings.isNullOrEmpty(offendingNetworkNames)) {
+                errors.add(ConfigValidationError.transitionError(
+                        String.format("PodSpec[name:%s]", newPod.getType()),
+                        "null",
+                        String.format("%s", newPod.getNetworks().toString()),
+                        String.format("New config has pod %s that uses network(s) %s, port mapping is " +
+                                        "not supported on these networks. NetworkSpec: %s", newPod.getType(),
+                                offendingNetworkNames, newPod.getNetworks().toString())));
+            }
+            // now we check that the none of the new pods move from not using host ports to using them (or vice
+            // versa)
+            PodSpec oldPod = oldPods.get(kv.getKey());
+            if (oldPod != null && podSpecUsesHostPorts(oldPod) != podSpecUsesHostPorts(newPod)) {
+                errors.add(ConfigValidationError.transitionError(
+                        String.format("PodSpec[name:%s]", oldPod.getType()),
+                        String.format("%s", oldPod.getNetworks().toString()),
+                        String.format("%s", newPod.getNetworks().toString()),
+                        String.format("New config has pod %s moving networks from %s to %s, changing it's " +
+                                        "host ports requirements from %s to %s, not allowed.",
+                                newPod.getType(), oldPod.getNetworks().toString(), newPod.getNetworks().toString(),
+                                podSpecUsesHostPorts(oldPod), podSpecUsesHostPorts(newPod))));
+            }
+        }
+        return errors;
+    }
+
+    private boolean podSpecUsesHostPorts(PodSpec podSpec) {
+        if (podSpec.getNetworks().size() == 0) {  // using HOST network, uses ports
+            return true;
+        }
+        for (NetworkSpec networkSpec : podSpec.getNetworks()) {
+            if (DcosConstants.networkSupportsPortMapping(networkSpec.getName())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String podSpecOverlayNetworkIsCorrect(PodSpec podSpec) {
```

Comment:
- Could be `static`?
- Maybe just return a `List<String>`
- Could be named something like `getNetworksWithInvalidPortMappings()`